src/test/integration-tests/integration-tests.ts

```diff
@@ -0,0 +1,172 @@
+            expect(mockHttpJSONRequest.parent.neverCalledWith(sinon.match({ baseUrl: "https://gaia.cruxpay.com?fakeKey=${console.log('test')}" }))).to.be.true;
+        })
+        it("resolving address of a user whose gaiaHub is tampered (malicious gaiaHub info response 2)", async () => {
+            const requestOptions = {
+                baseUrl: "https://hub.cruxpay.com",
+                url: "/hub_info",
+            };
+            const mockGaiaHubInfoResponse = {"challenge_text":"[\"gaiahub\",\"0\",\"hub.cruxpay.com\",\"blockstack_storage_please_sign\"]","latest_auth_version":"v1","max_file_upload_size_megabytes":20,"read_url_prefix":"https://gaia.cruxpay.com/f@akeHubPath!"};
+            const mockHttpJSONRequest = sandbox.stub(utils, "httpJSONRequest").callThrough().withArgs(sinon.match(requestOptions)).resolves(mockGaiaHubInfoResponse);
+            let raisedError;
+            try {
+                const resolvedAddress = await cruxWalletClient.resolveCurrencyAddressForCruxID(testCruxId.toString(), "btc");
+            } catch (error) {
+                raisedError = error;
+            }
+            expect(raisedError).to.be.instanceOf(CruxClientError);
+            expect(raisedError.message).to.include.string("invalid characters");
+            // @ts-ignore
+            expect(mockHttpJSONRequest.parent.neverCalledWith(sinon.match({ baseUrl: "https://gaia.cruxpay.com/f@akeHubPath!" }))).to.be.true;
+        })
+    })
+    describe("CruxWalletClient with testPrivateKey (registered)", () => {
+        beforeEach(() => {
+            // initialisation
+            const cruxWalletClientOptions: ICruxWalletClientOptions = {
+                walletClientName: "cruxdev",
+                privateKey: "L3ugweKa2xz4hbe7528qmm3Dvr9xEhHw7zdhk9zRgtRhmn1nrcHB",
+            }
+            cruxWalletClient = new CruxWalletClient(cruxWalletClientOptions);
+        })
+        it("get testCruxId registration status", async () => {
+            const cruxIdState = await cruxWalletClient.getCruxIDState();
+            expect(cruxIdState).to.be.eql({
+                cruxID: testCruxId.toString(),
+                status: {
+                    status: SubdomainRegistrationStatus.DONE,
+                    statusDetail: SubdomainRegistrationStatusDetail.DONE,
+                }
+            });
+        })
+        it("getAddressMap for the testCruxId", async () => {
+            const addressMap = await cruxWalletClient.getAddressMap();
+            expect(addressMap).to.be.eql({"btc": btcAddress});
+        })
+        it("putAddressMap for the testCruxId", async () => {
+            const {success, failures} = await cruxWalletClient.putAddressMap({"btc": btcAddress});
+            expect(success).to.be.eql({"btc": btcAddress});
+            expect(failures).to.be.eql({});
+        })
+        it("getAddressMap for the testCruxId when the gaia record is tampered", async () => {
+            const requestOptions = {
```

Comment:
Doesn't the CruxWalletClient creation fails if we gaia record are tampered for user.

Reply:
no, since the user entity is fetched in the scope of each method now.